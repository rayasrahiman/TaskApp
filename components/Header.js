import React from 'react';
import {
  View,
  Text,
  Image,
  StyleSheet,
  Dimensions,
  TouchableOpacity,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {Colors} from '../constants/colors';

export default function Header({
  dashboard,
  headerRight,
  onPress,
  title,
  ...headerProps
}) {
  return (
    <View
      {...headerProps}
      style={{
        ...styles.container,
        ...(dashboard && {
          marginTop: Dimensions.get('window').height < 800 ? -hp(11) : -hp(5),
        }),
      }}>
      <View style={styles.imgCont}>
        <Image
          source={require('../assets/Logo.png')}
          style={styles.img}
          resizeMode="contain"
        />
      </View>
      {headerRight && (
        <TouchableOpacity onPress={onPress} style={styles.btnCont}>
          <Text style={styles.btnTxt}>{title}</Text>
        </TouchableOpacity>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    // justifyContent: 'center',
    alignItems: 'center',
    borderBottomColor: Colors.gray,
    borderBottomWidth: wp(0.3),
    width: Dimensions.get('window').width,
    height: hp(15),
  },
  imgCont: {
    marginLeft: wp(40),
  },
  img: {
    width: wp(15),
    height: hp(15),
  },
  btnCont: {
    marginRight: wp(5),
  },
  btnTxt: {
    fontSize: hp(3),
    color: Colors.lightBlue,
  },
});
