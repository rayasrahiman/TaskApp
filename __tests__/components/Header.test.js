import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import Header from '../../components/Header';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('Header', () => {
  test('should render Header correctly', () => {
    const wrapper = render(<Header />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
