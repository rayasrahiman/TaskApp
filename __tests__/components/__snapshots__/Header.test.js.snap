// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#CCCCCC",
      "borderBottomWidth": 2.5,
      "flexDirection": "row",
      "height": 200,
      "justifyContent": "space-between",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 300,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/Logo.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 112.5,
        }
      }
    />
  </View>
</View>
`;
